### PR TITLE
Add/lstm direction

### PR DESCRIPTION
**Pull request by singjc**
Hi,

I added a direction field to the `LSTMConfig` using a enum for forward and reverse direction. I added this so that it's possible to retrieve the weights from reversed layers of a pretrained pytorch model. This also addresses issue #2351.

### CODE REVIEW

File: candle-nn/src/rnn.rs [hunk 1]
```diff
@@ -70,6 +70,12 @@ impl LSTMState {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+pub enum Direction {
+    Forward = 1,
```
Comment (LaurentMazare) [inline candle-nn/src/rnn.rs:75]: What is the need for associating integer values to this enum?

File: candle-nn/src/rnn.rs [hunk 2]
```diff
@@ -128,25 +137,29 @@ pub fn lstm(
     vb: crate::VarBuilder,
 ) -> Result<LSTM> {
     let layer_idx = config.layer_idx;
+    let direction_str = match config.direction {
+        Direction::Forward => String::from(""),
```
Comment (LaurentMazare) [inline candle-nn/src/rnn.rs:141]: Why use `String` rather than `&'static str` here?

File: candle-nn/src/rnn.rs (hunk 1 above)
Comment (singjc) [reply candle-nn/src/rnn.rs]: There actually isn't a need to assign an integer value. I removed it now

File: candle-nn/src/rnn.rs (hunk 2 above)
Comment (singjc) [reply candle-nn/src/rnn.rs]: Thanks for pointing that out. I'm still new to Rust, so wasn't quite sure about using `String` vs just having `&'static str'. But I see its probably more efficient to use `&'static str`. I updated the code per your suggestion.

**Review by LaurentMazare — APPROVED** [review]

**Comment by LaurentMazare** [comment]
Thanks!

**Comment by LaurentMazare** [comment]
Btw if you already have some models that use a bidirectional LSTM, would be great to have them as potential examples in candle-examples.

**Pull request merged**